File: VB/DXExampleVB/DXExample.Module/HideRolesViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl

Namespace DXExample.Module
	Partial Public Class HideRolesViewController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			TargetViewType = ViewType.ListView
			TargetObjectType = GetType(CustomRole)
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			Dim adminRole As CustomRole = View.ObjectSpace.Session.FindObject(Of CustomRole)(New BinaryOperator("Name", "Administrators"))
			If (TryCast(SecuritySystem.CurrentUser, CustomUser)).Roles.Lookup(adminRole.Oid) Is Nothing Then
				Dim listView As ListView = CType(View, ListView)
				listView.CollectionSource.Criteria("ByOwning") = CriteriaOperator.Parse("IsOwningRequired = false OR [Users][Oid = ?]", SecuritySystem.CurrentUserId)
			End If
		End Sub
	End Class
End Namespace

File: CS/Solution/DXExample.Module/HideRolesViewController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace DXExample.Module {
    public partial class HideRolesViewController : ViewController {
        public HideRolesViewController() {
            InitializeComponent();
            RegisterActions(components);
            TargetViewType = ViewType.ListView;
            TargetObjectType = typeof(CustomRole);
        }
        protected override void OnActivated() {
            base.OnActivated();
            CustomRole adminRole = View.ObjectSpace.FindObject<CustomRole>(new BinaryOperator("Name", "Administrators"));
            if ((SecuritySystem.CurrentUser as CustomUser).Roles.Lookup(adminRole.Oid) == null) {
                ListView listView = (ListView)View;
                listView.CollectionSource.Criteria["ByOwning"] = CriteriaOperator.Parse("IsOwningRequired = false OR [Users][Oid = ?]", SecuritySystem.CurrentUserId);
            }
        }
    }
}